### ISSUE TITLE
Review Ticket for 'Intro to PowerShell'

### ISSUE THREAD

**Issue by jerielizabeth** [programminghistorian/ph-submissions#18]
The Programming Historian has received a new tutorial: Intro to PowerShell by @tedawson. This lesson can be read at:

http://programminghistorian.github.io/ph-submissions/lessons/intro-to-powershell.html

The lesson is currently under editorial review. I will announce on this issue when it is open for public peer review.

---

The Programming Historian embraces openness in its review process, and endeavors to keep the review conversation here on GitHub. If anyone feels the need to discuss anything privately, you are welcome to email me. You can always turn to @ianmilligan1 or @miriamposner if you feel there's a need for an ombudsperson to step in.

Anti-Harassment Policy
_

> This is a statement of the Programming Historian's principles and sets expectations for the tone and style of all correspondence between reviewers, authors, editors, and contributors to our public forums.
> 
> The Programming Historian is dedicated to providing an open scholarly environment that offers community participants the freedom to thoroughly scrutinize ideas, to ask questions, make suggestions, or to requests for clarification, but also provides a harassment-free space for all contributors to the project, regardless of gender, gender identity and expression, sexual orientation, disability, physical appearance, body size, race, age or religion, or technical experience. We do not tolerate harassment or ad hominem attacks of community participants in any form. Participants violating these rules may be expelled from the community at the discretion of the editorial board. If anyone witnesses or feels they have been the victim of the above described activity, please contact our ombudspersons (Ian Milligan or Miriam Posner - http://programminghistorian.org/project-team). Thank you for helping us to create a safe space.

**Comment by jerielizabeth** [comment]
@tedawson, thank you for this great tutorial! I really like the way you've paced the explanations and the exercises -- it reads very easily and clearly. I think this will be a much used tutorial, and I know many Windows people who will appreciated instructions for their platform of choice.

A few stylistic things for you. The syntax for the links and the headers seems to be off. I think you need a space between the `#`'s and the titles for the headers and one less space between the [text] and (url) for the links. 

In the section, "Infinite Loops and Aborting with control-c," the last paragraph gets a bit long and your tone shifts a bit. As opposed to the framing of "if you have any experience programming", I would suggest shifting to "As you gain more experience working with the command line, you will discover ..." Something along those lines would better match your tone throughout the rest of the tutorial.

The only place where I am uncertain is the discussion of Python and environmental variables. You are doing a lot in this tutorial (all of which is excellent!), but in many ways the environmental variables introduces a whole additional layer of complication that goes beyond what users of this lesson are likely anticipating. I am happy to be convinced otherwise (and if convinced, we can debate the value of Python2 vs Python3), but that might be a section we could split off into another tutorial or as a  supplement to existing tutorials.

Excellent job!

**Comment by tedawson** [comment]
Hey @jerielizabeth, thanks for your comments! I've fixed the syntax issues - for some reason they displayed correctly on GitHub, but not on the submission website. Now they display correctly both places!

You were definitely right about the over-long paragraph about infinite loops. Following your suggestion, I just deleted everything from "If you have any experience programming" and replaced it with "As you gain experience working on the command line, you will often be flummoxed by your computer's overly literal interpretations of your commands, but you will also gain a feel for how to give it instructions it can follow. Infinite loops should be avoided at all costs, but they will occur, and when they do, just remember `control-c`." Still a bit long, but much shorter, and, I hope, less of a tone change.

On the final issue - I definitely agree that the entire third section feels a little out of place. It was important to me to include it, because I know from experience that getting Python set up to work properly on Windows can be **extremely** frustrating. Since Python seems to be the "house language" for _The Programming Historian_, I thought it made sense to make sure that, by the end of the tutorial, readers would be able to get it set up. But this section is definitely too much too quickly. I think the general discussion of setting environmental variables could definitely go - it's probably not that helpful to say "Take this template that hasn't been explained and change some stuff". Maybe the whole thing could be worked into the conclusion, along the lines of: "To get some command-line tools working properly, you may have to follow the advice on this TechNet page" and including the link already present? Maybe it could just be a brief note at the end about using Python - no discussion of which version or why, just something like "Many tutorials use Python. To get Python set up, you will have to enter this ... for some idea of what this does (TechNet link)". Alternatively, I could see providing a more exhaustive discussion of setting environmental variables and specific instructions for not only Python but lots of other programs - perhaps as a separate tutorial. That would require some time, of course! I haven't made any change yet, because while I definitely agree change is needed, I'm not quite sure what change to make.

**Comment by tedawson** [comment]
@jerielizabeth: After a little reflection, I decided to go with a sort of conservative fix for the time being - I drastically reduced the third section, leaving a short explanation of getting Python set up, but removing all the broader stuff about environment variables. I think it's a lot better this way. Let me know what you think!

**Comment by jerielizabeth** [comment]
@tedawson: My apologies for the delay -- I had some unexpected travel last week and am just getting my feet under me again. 

I think the conservative fix works well. It signposts the broader issues well and provides enough to get someone started, without loosing them in the details. And the other changes look good to me as well!

I think we are ready to open this up for review! What say you?

**Comment by tedawson** [comment]
@jerielizabeth: no worries on the delay! Opening it up for review sounds good to me. Thanks for all of your suggestions and help thus far. I look forward to seeing what others say.

**Comment by jerielizabeth** [comment]
This lesson is now under review and can be read at:

http://programminghistorian.github.io/ph-submissions/lessons/intro-to-powershell.html

I will act as editor for the review process. My role is to solicit two reviews from the community and to manage the discussions, which should be held here on this forum. I have already read through the lesson and provided feedback, to which the author has responded.

Members of the wider community are also invited to offer constructive feedback which should post to this message thread, but they are asked to first read our Reviewer Guidelines (http://programminghistorian.org/reviewer-guidelines) and to adhere to our anti-harassment policy (below). We ask that all reviews stop after the second formal review has been submitted so that the author can focus on any revisions. I will make an announcement on this thread when that has occurred.

I will endeavor to keep the conversation open here on Github. If anyone feels the need to discuss anything privately, you are welcome to email me. You can always turn to @ianmilligan1 if you feel there's a need for an ombudsperson to step in.

> Anti-Harassment Policy
> _
> 
> This is a statement of the Programming Historian's principles and sets expectations for the tone and style of all correspondence between reviewers, authors, editors, and contributors to our public forums.
> 
> The Programming Historian is dedicated to providing an open scholarly environment that offers community participants the freedom to thoroughly scrutinize ideas, to ask questions, make suggestions, or to requests for clarification, but also provides a harassment-free space for all contributors to the project, regardless of gender, gender identity and expression, sexual orientation, disability, physical appearance, body size, race, age or religion, or technical experience. We do not tolerate harassment or ad hominem attacks of community participants in any form. Participants violating these rules may be expelled from the community at the discretion of the editorial board. If anyone witnesses or feels they have been the victim of the above described activity, please contact our ombudspersons (Ian Milligan - http://programminghistorian.org/project-team). Thank you for helping us to create a safe space.
> _

**Comment by erinbush** [comment]
@tedawson, thank you for a great Windows tutorial. I'm in agreement with Jeri that the explanations and exercises read easily and are clear and fun to follow. The flow and pacing is great and you build through the commands in a logical fashion. Additionally, the tone is great throughout. In reviewing and testing it out, I do have a couple of suggestions -- mostly to add more clarity. Please let me know if anything here doesn't make sense or if I lose you. I'll note the section first, hopefully to help with that.

**What/Why** - I actually think it might be helpful to list one or two examples of the dh tools that require the command line as you open. You do mention a couple at the end of the tutorial, but opening with something akin to your  first graf of "Running Python Scripts" could help those users who aren't sure they actually need this; they may change their minds upon seeing names of tools they recognize.

**Creating new directories** - I suggest linking to something that explains CamelCase in case it's an unfamiliar term. You might also want to spell out that "home" equals the userdirectory (where PS starts on open), as some beginners used to a GUI may come into this believing their Desktop is "home." And a random fun tip: on Tab to autocomplete, repeated hitting of the Tab button will actually cycle through all the directories -- you don't even need to type a letter after sl to get it started -- I'm pretty tickled that this worked and I feel like I found a $5 bill in my pocket. (Humor me, please!)

**Moving fast** -  You mention that pushd and popd are useful when you're moving frequently between two directories, but you may want to clarify that you'll have to type out the pushd string each time you want to get back to your working directory. One could misinterpret that popd functions to cycle between two recent directories.

**Moving files** - It might be important to note here that when you're in a subdirectory mv will always rename and that actually moving files from one subdir to another will require adding a redirection command, which you'll cover next.

I didn't see the original Python section in question in the above comments, but I think that section in particular read fine. Just enough information here to get things running smoothly.

Overall, this tutorial gave me a much-needed refresher and actually solidified some concepts that were previously confusing. Great job!

**Comment by tedawson** [comment]
@erinbush, thanks for your review! As I understand it, I should wait for a second review before actually undertaking any changes, but for now: all of your suggestions are very helpful. Thank you! And I didn't know the Tab "cycling trick" - it's awesome - I feel like you just gave me $5!

**Comment by jerielizabeth** [comment]
@erinbush Thank you so much for the review! These are really great suggestions! 

We are still working on soliciting the second review, but once that is complete, I'll add my summary suggestions and will close the review stage. In the meantime, any additional comments from the PH reviewing audience are welcome!

**Comment by derektprice** [comment]
@tedawson Great lesson. I was actually fairly new to Powershell before this tutorial, having only used it briefly as part of an intro to a Python tutorial that I did last summer, so I might actually be a decent representative of someone coming to the lesson with very little prior experience. A couple of quick notes:
- For people new to the CLI, it might be helpful the first time you mention directories to compare them to folders, something beginner will be familiar with (never mind, you do that later in the explorer section. Although it still might be better to do earlier?)
- In the "Moving Between Directories with Set-Location (sl, cd)" section, you might want to change "Navigate back to the desktop, and then back again to Users." to something like "Now let's try navigating back to the desktop, and then back again to Users." Small thing, but the imperative make it sound like I should be able to do this already.
- great explanation of pushd and popd - finally understand a bit more how they work!
- not really a suggestion, but I just really liked doing gc on "bigben.txt" and getting a wall of text about reproduction and plurality. It's a nice touch.
- that said, I wonder if there would be a better way to do the concatenation example: perhaps doing it in a different directory, putting a single word in each file, and then producing an end-result file that forms a complete sentence. It seems like something along those lines would allow someone to more easily know whether or not they've actually concatenated the files correctly. Admittedly, this would require re-writing that whole section, which is a big change, but I just thought I'd throw the idea out there. It's a little "natural language" check that might help those struggling to adapt to "computer language"
- final suggestion: My sense, having finished the tutorial, is that this is intended for someone just starting out, almost an absolute beginner. It might be nice to mention that up front, as that tends to set people at ease, and lets the reader know if this tutorial is for them or not.

Besides these few comments, I think the whole tutorial flows really well, has an appropriate tone, and opens up the possibilities of PowerShell without overloading of bombarding the reader with detail. Great work!

**Comment by tedawson** [comment]
Hey @derektprice, thanks for the feedback! This is really helpful, and I'm glad you appreciated the little "joke" with making reproductions of the Benjamin text!

**Comment by jerielizabeth** [comment]
Hi everyone!

Thank you all for the very thorough and helpful feedback! The question of the concatenation section is an interesting one and I am not sure that I have a strong editorial opinion either way. I would encourage talking some more here in the comments about the pros and cons of changing that section.

I will leave this issue open so that the conversation can continue as @tedawson revises the lesson. Once the lesson is completed and published in the main repository, I will close the issue.

**Comment by tedawson** [comment]
Hi all!

Before I set to making revisions, I thought I'd share my plans for how to respond to your reviews.

@erinbush: Basically, yes to all. I think all of your points are good, and will allow me to make minor tweaks that will improve the tutorial.

@derektprice: to your first two points (folder/directory, "let's try" rather than "do this"), same as above. The issue with concatenation is a bit of a doozy, though, and I would love any thoughts any of you have on this. Here's what I'm thinking at the moment:

There are things I like about the current example, and it's somewhat integral to the flow of the lesson, as it sets up the discussion of parameters. As such, I don't want to remove it. I actually don't think I ever really set out to explain concatenation, and maybe that's the issue. What would seem to me the easiest thing to do without disrupting the flow would be to change the **reading files with get-content** section so that rather than creating a new file with a second sentence in it, you create several new files with parts of that sentence in them and then concatenate them into one file. By doing this, I think I could better explain concatenation without disrupting the current flow. But I also think this would be a rather long activity - several writes, then the concatenation, then removing the extra files so that the next section functions properly. To me, then, the question becomes: Is it worth it? I really want to find that perfect balance of explaining everything with just enough examples that it's not confusing, but without too many so that the reader gets bogged down, and I wonder if this wouldn't push the tutorial in the too-much direction. Would love everyone's thoughts on this.

Finally, to @derektprice's last point: Yes, this is absolutely meant for beginners. As far as I know, a note on difficulty is usually included at the top of every lesson. @jerielizabeth, I assume this is part of the YAML block? Do I need to add this?

Thanks again to everyone!

**Comment by derektprice** [comment]
@tedawson - Concerning the length and the question of re-writing: I think your sense that any additional material would make it a bit too long are on point. As I mentioned at the end, I really do think the tutorial has a nice flow as it is, and it could be that trying to re-write that single section would actually muddy the waters more than anything else, especially since explaining concatenation isn't really your task here. Reading over the section again, I do feel that it actually serves its purpose as-is, so basically I'd leave it up to you to decide what you'd like to do with that section, unless anyone else has a strong opinion on this.

Thanks!

**Comment by jerielizabeth** [comment]
I agree that tutorial currently flows nicely and strikes a good balance between too much and too little information. I think the "reading files with get-content" section does a lot of important work, so I am hesitant of the idea of reworking it to emphasize concatenating. In fact, reading back through those sections, it might be better to link to an external resource on concatenation (or file manipulation in general) for those who really want to know more. 

@tedawson about the difficulty level -- I think that does get set in the yaml and I'll take care of that part as I need to look up how we're currently handling that.

**Comment by jerielizabeth** [comment]
Also, this tutorial is already getting incorporated into the Programming Historian ecosystem -- the upcoming tutorial on JSON and Jq will be linking here! :)

**Comment by tedawson** [comment]
Awesome news @jerielizabeth! Thanks to everyone for your feedback. On the concatenation, I think I'll go with the 'link something' solution. This week is pretty hectic, but I hope by the end of it to have this and all the other revisions made.

**Comment by jerielizabeth** [comment]
Hi all! I, your mac-using tour guide, have a quick favor to ask! Would one of you be willing to check to see if the instructions for installing jq via PowerShell work? The instructions are here: [http://programminghistorian.github.io/ph-submissions/lessons/json-and-jq#installation-on-windows](http://programminghistorian.github.io/ph-submissions/lessons/json-and-jq#installation-on-windows) and any comments you have can be added here: [https://github.com/programminghistorian/ph-submissions/issues/23](https://github.com/programminghistorian/ph-submissions/issues/23).

**Comment by tedawson** [comment]
Hi everyone! Finally responded to all the reviewer feedback. See the change log below. Mostly this is all pretty straight-forward, but there is one change that I found myself really second guessing (in bold to make obvious). Please give me your thoughts. Once that is straightened out, I think this is all ready to go.

Change Log:

-moved explicit naming of command-line tools with links to other PH tutorials to beginning (What/Why) rather than end (using command-line tools).

-in "Navigation", added early clarification of "directory = folder" and home being users folder, not desktop

-changed "Navigate back" to "Try navigating back" in "moving between directories" section, and added slight clarification

-in "Deletion with Remove-Item" added use of tab and shift+tab for scrolling through all available directories (made more sense to do it here than in initial discussion of auto-completion.) (Thanks @erinbush)

-added explanation of what pushd and popd do under "Moving Fast", including a link to the wikipedia article on the call stack

-under "copying and moving" added this paragraph: 
How is it that `mv` can both move and rename files? Every time we use `mv` we give it two arguments, that is, we type two things after the `mv`. The first thing is the path for the original item, the second thing is the path for the destination we want to move the item to. In both cases, PowerShell automatically completes the path information we leave out, and it assumes that the title will remain the same if we don't tell it otherwise. So the first command really looks like this `mv .\example.txt .\dir\example.txt`. In the second case, since there is no directory called `benjamin.txt`, the whole command looks like this: `mv .\example.txt .\benjamin.txt`. In other words, it is moving the file from the path `C:\users\YOURUSERNAME\funWithPowerShell\dir\example.txt` to the path  `C:\users\YOURUSERNAME\funWithPowerShell\dir\example.txt`. That is, renaming _is_ moving (which is also true when you use your GUI, but the GUI magic makes it less apparent).
**I'm not really sure this is an improvement! I'm attempting to respond to @erinbush's point that the discussion of `mv` might be misleading, but I think explaining it might be too much information... Incidentally, I don't think you need redirection to move between subdirectories - if you're in directory/subdirectory and want to move x.txt to directory/subdirectory1, you just type `mv x.txt ../subdirectory1`. If you don't type the .. it'll look for subdirectory1 in subdirectory and, not finding it, assume change x.txt's path to directory/subdirectory/subdirectory1, where "subdirectory1" is now a text file. I'm at a loss for a way to elegantly explain this, though, and am thinking it might require a more substantial rewrite that really dives into what a path is and how your computer finds things. Thoughts?**

-Under 'solving the infinite loop problem' added the following paragraph to address the question of learning more about concatenation: 
Throughout this process, we've been adding texts together, or concatenating them. You can learn more about [concatenation from Wikipedia](https://en.wikipedia.org/wiki/Concatenation), and if you want to see a few more examples of concatenation using PowerShell, check out this [blog entry](https://blogs.technet.microsoft.com/heyscriptingguy/2014/07/15/keep-your-hands-clean-use-powershell-to-glue-strings-together/#comments), which will take you into the wonderful world of variables, something beyond the scope of this tutorial, but well worth learning about.

-Added Links to the following: 
CamelCase page on Wikipedia
Call Stack page on Wikipedia
Concatenation page on Wikipedia
TechNEt blog about concatenation

**Comment by tedawson** [comment]
@jerielizabeth and @ianmilligan1: regarding the jq thing - I'd be happy to do this! BUT between now and July 1st, I am moving back to the States from Austria, going to a family wedding in Michigan, and then moving to a new home in Nashville, so it'll be a little hectic, and the earliest I can promise actually doing this would be the beginning of July...

**Comment by erinbush** [comment]
@tedawson  I agree with you.. the explanation of mv is too much information and that wasn't my original intent. My note was really to see if clarifying the functionality was a possibility.  Would it be useful to add a line simply stating that mv can both add and rename files, so be aware of what you're doing--and leave it at that? Just as a warning. I don't want to make it more complicated.

**Comment by ianmilligan1** [comment]
@tedawson No worries on the `jq` front – I can find a Windows box at some point this week while at an event. Good luck with your move(s) and enjoy the wedding (having just been at a family wedding in Michigan, I can imagine that you'll have fun!).

**Comment by tedawson** [comment]
@erinbush: okay, so I changed it to the following:

You may be surprised that the same commandlet is used both to move and to rename files. In fact, though, the operation is the same. In both cases, you're telling the computer to change the "name" of the file's location. In the first example above, the file's "name" began as 

`C:\Users\Ted\funWithPowerShell\example.txt` 

and changed to 

`C:\Users\Ted\funWithPowerShell\dir\example.txt`

In the second example, the "name" changed from 

`C:\Users\Ted\funWithPowerShell\dir\example.txt` 

to

`C:\Users\Ted\funWithPowerShell\dir\benjamin.txt` 

We will see below that this "name" is actually called a path, and we'll learn a little more about how it works. For now, just be careful when you use `mv`, because if you don't type exactly the right thing, you may rename something when you mean to move it, or vice versa.

Still a bit long, perhaps, but I think it gets the point across without too much extra(neous) information...

**Comment by tedawson** [comment]
@erinbush, so I just changed it again, because I realized on reading over the entire lesson that I'd already explained what the path is. Below is the current rendering; let me know what you think. @jerielizabeth: assuming this last change is good, I think all the reviewer suggestions have been responded to!

You may be surprised that the same cmdlet is used both to move and to rename files. In fact, though, the operation is the same. In both cases, you're telling the computer to change the "name" of the file's location, that is, to change the path it follows to find the file. In the first example above, the path began as

C:\Users\Ted\funWithPowerShell\example.txt

and changed to

C:\Users\Ted\funWithPowerShell\dir\example.txt

In the second example, the path changed from

C:\Users\Ted\funWithPowerShell\dir\example.txt

to

C:\Users\Ted\funWithPowerShell\dir\benjamin.txt

In other words, in both examples, mv just changed the path. Don't worry if this doesn't make sense, though. Just be careful when you use mv, because if you don't type exactly the right thing, you may rename something when you mean to move it, or vice versa.

**Comment by erinbush** [comment]
@tedawson I like that.. very clear. Thanks Ted. Great tutorial!

**Comment by jerielizabeth** [comment]
@tedawson Great! Thank you so much @erinbush and @derektprice for your reviews and subsequent comments! I'll read through for copy-editing hopefully this weekend.

I've added the reviewer and recommended user information to the tutorial. Let me know if you would like your name to display differently. Ted, can you send me a once sentence author bio so that I can add that to author yaml file, either here or via email?

**Comment by wcaleb** [comment]
The figure syntax for the images in this lesson need to be updated according to the [new guidelines posted here](http://programminghistorian.org/new-lesson-workflow#figures).

**Comment by jerielizabeth** [comment]
I updated figure syntax for the images. All is displaying well.

**Comment by jerielizabeth** [comment]
Two thoughts on the directory image for this lesson, @tedawson. Let me know which you prefer:

Image 1:
![intro-to-powershell](https://cloud.githubusercontent.com/assets/1447744/16972337/ea87ce5e-4e2a-11e6-8478-e502bb9ad4e0.png)

Image 2:

![intro-to-powershell-1](https://cloud.githubusercontent.com/assets/1447744/16972330/df8b069c-4e2a-11e6-962f-f13096a73555.png)

**Comment by tedawson** [comment]
Hi @jerielizabeth! I like them both, and am really awful with graphic design, so will defer to your judgment if you disagree - but - I'd go with 2!

**Comment by fredgibbs** [comment]
it's not for me to say, but please please go with the shell!

**Comment by tedawson** [comment]
Works for me! Let's do the shell!

**Comment by mdlincoln** [comment]
🐚 🐚 🐚

**Comment by jerielizabeth** [comment]
haha! I think the "shells" have it!

## 

Jeri E. Wieringa
Digital Publishing Production Lead, George Mason University Libraries
PhD Candidate, Department of History and Art History
George Mason University
www.jeriwieringa.com

**Comment by jerielizabeth** [comment]
Alright! The lesson has been moved over to the main blog for a "soft" publish. Last chance to catch any typos or corrections before I start promoting tomorrow! Congrats @tedawson on a successful and excellent tutorial!

**Comment by ianmilligan1** [comment]
Fantastic! We're also launching #23 in a similar manner.

**Comment by tedawson** [comment]
@jerielizabeth - bit of a problem - I was sitting at a computer at the Vanderbilt library, and thought I'd read through the lesson a last time. This computer has Windows 7 and PS 2.0 (current version is 5.0). To my horror, a couple of things from the lesson don't work correctly in 2.0. Nothing major. The New-Item cmdlet requires a parameter to work properly, the Select-String cmdlet does not have the "sls" alias used in the lesson, and one parameter with Measure-Object and one parameter with Get-Content work slightly differently. Someone using 2.0 could still go through the lesson with a couple of small frustrations, in other words. I doubt anyone still has this out-of-date of a PS version on their own computer, but it's possible people on University networks might (then again, Vanderbilt is pretty slow in IT stuff, so I really don't know how big an issue this might be). I wonder if we need to add a mention that the lesson was written using 5.0, and, if possible, people should make sure their software is up-to-date (and that if not possible, they may run into a few bump). Then again, perhaps we could just ignore it, since it doesn't effect the major content of the lesson, and the "if your technology hasn't been updated in 8 years this might not work" caveat is sort of a given.

I'm really embarrassed and sorry to not have seen this issue until just now.

I also noted one small typo in the caption to intro-to-powershell5.png, but wasn't sure whether I should fix it at this point.

**Comment by fredgibbs** [comment]
Although i don't think there's an official editorial policy about this, the practice seems to have converged on announcing early in the lesson what versions of software have been used, and that previous versions will usually work with some googling to account for variations in functionality (or to say that things won't work). Based on experience, it's better to be explicit about the versions used than to remain silent, because a few years from now we'll have forgotten what version the lesson is based on and there will be some small bug due to newer version functionality.

Most PH lessons tacitly assume that one is working on one's own computer and can install software and so on, so I don't think the PS version issue (more likely in a lab setting as you note) conflict with the general baseline PH assumptions for lessons generally.

**Comment by jerielizabeth** [comment]
Thanks, Fred!

I concur -- versions of software is a perennial problem with this sort of
work and it is much better to just state what we've written for and not try
to cover all possible previous versions. Let's add a note to the beginning
of the lesson stating that the lesson is written for PowerShell 5.0 and
note that some modifications will be necessary with earlier versions of PS.

Ted, if you want to make those modifications on the draft version in
ph-submissions, I'll re-upload from there to the jekyll repository.

Thanks!

## 

Jeri E. Wieringa
Digital Publishing Production Lead, George Mason University Libraries
PhD Candidate, Department of History and Art History
George Mason University
www.jeriwieringa.com

**Comment by tedawson** [comment]
Done! Thanks, @jerielizabeth for all of your help throughout the process!

**Comment by jerielizabeth** [comment]
my pleasure! The changes have been moved over!

## 

Jeri E. Wieringa
Digital Publishing Production Lead, George Mason University Libraries
PhD Candidate, Department of History and Art History
George Mason University
www.jeriwieringa.com

**Comment by jerielizabeth** [comment]
I think we are good to declare the "Intro to PowerShell" published! I will start promoting it on Twitter. The final lesson is now available at [http://programminghistorian.org/lessons/intro-to-powershell](http://programminghistorian.org/lessons/intro-to-powershell) and I will close out the review and editing process issue. 

Thank you again to our reviewers @erinbush and @derektprice for your incredibly helpful feedback! And congratulations once again to @tedawson!
